worlds/tunic/__init__.py

```diff
@@ -237,11 +244,21 @@ def create_items(self) -> None:
             items_to_create["Hero's Laurels"] = 0
 
         if self.options.keys_behind_bosses:
-            for rgb_hexagon, location in hexagon_locations.items():
-                hex_item = self.create_item(gold_hexagon if self.options.hexagon_quest else rgb_hexagon)
-                self.get_location(location).place_locked_item(hex_item)
-                items_to_create[rgb_hexagon] = 0
-            items_to_create[gold_hexagon] -= 3
+            rgb_hexagons = list(hexagon_locations.keys())
+            # shuffle these in case not all are placed in hex quest
+            self.random.shuffle(rgb_hexagons)
+            for rgb_hexagon in rgb_hexagons:
+                location = hexagon_locations[rgb_hexagon]
+                if self.options.hexagon_quest:
+                    if items_to_create[gold_hexagon] > 0:
+                        hex_item = self.create_item(gold_hexagon)
+                        items_to_create[gold_hexagon] -= 1
+                        items_to_create[rgb_hexagon] = 0
+                        self.get_location(location).place_locked_item(hex_item)
+                else:
+                    hex_item = self.create_item(rgb_hexagon)
+                    self.get_location(location).place_locked_item(hex_item)
+                    items_to_create[rgb_hexagon] = 0
```

Comment:
What happens here if you have hex quest on, and you only have 1 gold hexes in the pool when you start this loop? Would it never the `items_to_create[rgb_hexagon]` to 0 for 2 of them, or is that handled somewhere else?

Reply:
Any remaining ones should get handled normally when it goes to swap out the pages/hexagons (line 290); this change here is basically to avoid the edge case you described